lib/improver/cli/temp_lapse_rate.py

```diff
@@ -32,151 +32,91 @@
 
     Args:
-        temperature_cube (iris.cube.Cube):
-            A cube of air temperature to be processed (K).
-        orography_cube (iris.cube.Cube):
-             A Cube containing orography data (metres).
-        land_sea_mask_cube (iris.cube.Cube):
-            A cube containing a binary land-sea mask.
-            True for land-points.
-            False for sea.
-        max_height_diff (float):
+        temperature (iris.cube.Cube):
+            Air temperature data. This is required even when returning DALR,
+            as this defines the grid on which lapse rates are required.
+        orography (iris.cube.Cube):
+            Orography data.
+        land_sea_mask (iris.cube.Cube):
+            Binary land-sea mask data. True for land-points, False for sea.
+        max_height_diff_metres (float):
             Maximum allowable height difference between the central point and
             points in the neighbourhood over which the lapse rate will be
             calculated.
-            Default is 35.
         nbhood_radius (int):
-            Radius of neighbourhood around each point. The neighbourhood
-            will be a square array with side length 2*nbhood_radius + 1.
-            The default value of 7 is from the reference paper.
+            Radius of neighbourhood in grid points around each point. The
+            neighbourhood is a square array with side length
+            2*nbhood_radius + 1. The default value of 7 is from the reference
+            paper (see plugin documentation).
         max_lapse_rate (float):
-            Maximum lapse rate allowed.
-            Default is 3*improver.constants.DALR.
+            Maximum lapse rate allowed, in K m-1.
         min_lapse_rate (float):
-            Minimum lapse rate allowed.
-            Default is improver.constants.DALR.
+            Minimum lapse rate allowed, in K m-1.
         return_dalr (bool):
             If True, returns a cube containing the dry adiabatic lapse rate
             rather than calculating the true lapse rate.
 
     Returns:
         iris.cube.Cube:
-            Cube containing lapse rate (K m-1)
+            Lapse rate (K m-1)
 
     Raises:
-        ValueError:
-            If minimum lapse rate is greater than maximum.
-        ValueError:
-            If Maximum height difference is less than zero.
-        ValueError:
-            If neighbourhood radius is less than zero.
+        ValueError: If minimum lapse rate is greater than maximum.
+        ValueError: If Maximum height difference is less than zero.
+        ValueError: If neighbourhood radius is less than zero.
 
     """
+    import numpy as np
+    from improver.lapse_rate import LapseRate
+    from improver.metadata.amend import amend_attributes
+
+    attributes_dict = {"title": "remove", "source": "remove",
+                       "history": "remove", "um_version": "remove"}
+
+    if return_dalr:
```

Comment:
Since this is just a constant array, and we only pass the temperature in to find its shape, I think we should separate out the dry adiabatic lapse rate stuff into its own CLI (it's essentially an ancillary) - that way we can pass in the orography and land-sea mask as mandatory positional inputs, which fits with the general pattern we'd like to enforce.

What do you think?

Reply:
I think we probably don't need a CLI for this in the long term, but it is useful functionality to have for running scientific trials.  From my perspective it is easier to strip this statement out of an existing CLI - especially if we mark it as "temporary" or "deprecated" - than to remember to delete a whole CLI we don't need at a later date.  But I'm not sure whether we could actually get rid of it now.

@bayliffe what do you think?

Reply:
Leave this for now. We will either abandon local lapse rates and choose a sensible fixed lapse rate (if it shows skill), in which case this whole CLI will be canned. Or we will refine and use local lapse rates and no longer need the DALR option. Making changes now is not a useful way to spend time.

Reply:
Discussed this with @cgsandford. If we really need this functionality for now, I think the least worst approach is to have both the ancillaries as mandatory inputs, because they're needed for the actual lapse rate calculation - in the case where we're just returning a constant - it's a bit of a waste to pass these in, but it does help make the interface consistent with everything else we're doing, and unblocks other green team work. How do you feel about that, @bayliffe?
(cc @TomekTrzeciak)

Reply:
As long as we retain the DALR option for now I am happy.

Reply:
After discissing wtih @TomekTrzeciak, we decided to go with making these optional postional arguments, given that we either need both or neither.